crypto.go

```diff
@@ -364,8 +369,8 @@ func (sk PrivateKey) Derive(index uint32) (PrivateKey, error) {
 	return derivedKey, nil
 }
 
-func (sk PrivateKey) LegacyDerive(index int32) (PrivateKey, error) {
-	keyData := deriveLegacyChildKey(sk.keyData, index)
+func (sk PrivateKey) LegacyDerive(index int64) (PrivateKey, error) {
```

Comment:
This is technically a breaking change. Although I do understand we kind of need to do this. Hmm....

Reply:
if it was java would be easier here